src/initializers/disable-core-sitemaps.php

```diff
@@ -16,10 +17,84 @@ class Disable_Core_Sitemaps implements Initializer_Interface {
 
 	use No_Conditionals;
 
+	/**
+	 * The options helper.
+	 *
+	 * @var Options_Helper
+	 */
+	private $options;
+
+	/**
+	 * Sitemaps_Enabled_Conditional constructor.
+	 *
+	 * @param Options_Helper $options The options helper.
+	 *
+	 * @codeCoverageIgnore
+	 */
+	public function __construct( Options_Helper $options ) {
+		$this->options = $options;
+	}
+
 	/**
 	 * Disable the WP core XML sitemaps.
 	 */
 	public function initialize() {
 		\add_filter( 'wp_sitemaps_enabled', '__return_false' );
+
+		if ( $this->options->get( 'enable_xml_sitemap' ) ) {
+			\add_action( 'template_redirect', [ $this, 'template_redirect' ], 0 );
+		}
+	}
+
+	/**
+	 * Redirects requests to the WordPress sitemap to the Yoast sitemap.
+	 *
+	 * @return void
+	 */
+	public function template_redirect() {
+		// If there is no path, nothing to do.
+		if ( ! isset( $_SERVER['REQUEST_URI'] ) ) {
+			return;
+		}
+		$path = \sanitize_text_field( \wp_unslash( $_SERVER['REQUEST_URI'] ) );
+
+		// If it's not a wp-sitemap request, nothing to do.
+		if ( \substr( $path, 0, 11 ) !== '/wp-sitemap' ) {
+			return;
+		}
+
+		$redirect = $this->get_redirect_url( $path );
+
+		if ( ! $redirect ) {
+			return;
+		}
+
+		\wp_safe_redirect( \home_url( $redirect ), 301, 'Yoast SEO' );
+	}
+
+	/**
+	 * Returns the relative sitemap URL to redirect to.
+	 *
+	 * @param string $path The original path.
+	 *
+	 * @return string|false The path to redirct to. False if no redirect should be done.
+	 */
+	private function get_redirect_url( $path ) {
+		// Start with the simple string comparison so we avoid doing unnecessary regexes.
+		if ( $path === '/wp-sitemap.xml' ) {
+			return '/sitemap_index.xml';
+		}
+
+		if ( \preg_match( '/^\/wp-sitemap-(posts|taxonomies)-(\w+)-(\d+).xml$/', $path, $matches ) ) {
```

Comment:
The dot represents any character that is not a new-line. This will work because it will match a ".", but let's do this correctly.

```suggestion
		if ( \preg_match( '/^\/wp-sitemap-(posts|taxonomies)-(\w+)-(\d+)\.xml$/', $path, $matches ) ) {
```